src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4541,6 +4541,11 @@ public boolean isStarting()
         return operationMode == Mode.STARTING;
     }
 
+    public boolean isJoiningOrWaitingToFinishBootstrap()
+    {
+        return operationMode == Mode.JOINING || operationMode == Mode.WAITING_TO_FINISH_BOOTSTRAP;
```

Comment:
should we make operationMode volatile?

Reply:
looks like mainline does: https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/service/StorageService.java#L454

Reply:
Yeah it was changed in https://issues.apache.org/jira/browse/CASSANDRA-12509 to improve the shutdown procedure. I've pushed a change to make opmode volatile